packages/editor-ui/src/stores/tags.store.ts

```diff
@@ -1,105 +1,119 @@
-import { createTag, deleteTag, getTags, updateTag } from '@/api/tags';
+import * as tagsApi from '@/api/tags';
 import { STORES } from '@/constants';
-import type { ITag, ITagsState } from '@/Interface';
+import type { ITag } from '@/Interface';
 import { defineStore } from 'pinia';
 import { useRootStore } from './root.store';
+import { computed, ref } from 'vue';
 import { useWorkflowsStore } from './workflows.store';
 
-export const useTagsStore = defineStore(STORES.TAGS, {
-	state: (): ITagsState => ({
-		tags: {},
-		loading: false,
-		fetchedAll: false,
-		fetchedUsageCount: false,
-	}),
-	getters: {
-		allTags(): ITag[] {
-			return Object.values(this.tags).sort((a, b) => a.name.localeCompare(b.name));
-		},
-		isLoading(): boolean {
-			return this.loading;
-		},
-		hasTags(): boolean {
-			return Object.keys(this.tags).length > 0;
-		},
-		getTagById() {
-			return (id: string) => this.tags[id];
-		},
-	},
-	actions: {
-		setAllTags(tags: ITag[]): void {
-			this.tags = tags.reduce((accu: { [id: string]: ITag }, tag: ITag) => {
-				accu[tag.id] = tag;
-
-				return accu;
-			}, {});
-			this.fetchedAll = true;
-		},
-		upsertTags(tags: ITag[]): void {
-			tags.forEach((tag) => {
-				const tagId = tag.id;
-				const currentTag = this.tags[tagId];
-				if (currentTag) {
-					const newTag = {
-						...currentTag,
-						...tag,
-					};
-					this.tags = {
-						...this.tags,
-						[tagId]: newTag,
-					};
-				} else {
-					this.tags = {
-						...this.tags,
-						[tagId]: tag,
-					};
-				}
-			});
-		},
-		deleteTag(id: string): void {
-			const { [id]: deleted, ...rest } = this.tags;
-			this.tags = rest;
-		},
-
-		async fetchAll(params?: { force?: boolean; withUsageCount?: boolean }): Promise<ITag[]> {
-			const { force = false, withUsageCount = false } = params || {};
-			if (!force && this.fetchedAll && this.fetchedUsageCount === withUsageCount) {
-				return Object.values(this.tags);
-			}
+export const useTagsStore = defineStore(STORES.TAGS, () => {
+	const tags = ref<{ [id: string]: ITag }>({});
+	const loading = ref(false);
+	const fetchedAll = ref(false);
+	const fetchedUsageCount = ref(false);
+
+	const rootStore = useRootStore();
+	const workflowsStore = useWorkflowsStore();
+
+	// Computed
+
+	const allTags = computed(() => {
+		return Object.values(tags.value).sort((a, b) => a.name.localeCompare(b.name));
+	});
+
+	const isLoading = computed(() => loading.value);
+
+	const hasTags = computed(() => Object.keys(tags.value).length > 0);
+
+	const getTagById = computed(() => {
+		return (id: string) => tags.value[id];
+	});
```

Comment:
Computed getters will break in composition API if used within another computed value. We should just rename `tags` to `tagsById`, and reference the value directly instead (i.e. `tagsStore.tagsById[id]`).